kernel/indTyping.ml

```diff
@@ -286,10 +307,14 @@ let typecheck_inductive env (mie:mutual_inductive_entry) =
   mind_check_names mie;
   assert (List.is_empty (Environ.rel_context env));
 
+  let has_template_poly = List.exists (fun oie -> oie.mind_entry_template) mie.mind_entry_inds in
+
   (* universes *)
   let env_univs =
     match mie.mind_entry_universes with
-    | Monomorphic_entry ctx -> push_context_set ctx env
+    | Monomorphic_entry ctx ->
+      let env = if has_template_poly then set_universes_lbound env Univ.Level.prop else env in
+      push_context_set ctx env
```

Comment:
push_context_set catches the AlreadyDeclared exception, given that afterwards we assume the universes in ctx are new this is problematic.
We hope to someday stop catching this exception but sideeffect intricacies prevent it for now, so I guess adding an optional argument `?enforce_fresh` (defaulting with current behaviour) should be added to push_context_set

Reply:
You’re right I didn’t catch that one